### PR TITLE
Initial TemplateCategory client

### PR DESCRIPTION
**Pull request by whabanks**
# Summary | Résumé

This PR implements the `template_category_api_client`, implementing the following functionality on the platform admin pages.

- Get list of Categories
- Get data for a single Category
- Update a Category
- Create a new Category

# Related issues
* https://github.com/cds-snc/notification-api/pull/2208

# Test instructions | Instructions pour tester la modification
1. Navigate to the `Template Categories` platform admin page - Note that the three generic categories are present
2. Click the `New Category` button - Fill out the form and create the category - Note you're taken back to the list of template categories page
3. Select the category you just created - Note that the form is filled correctly with the data you input on step 2
4. Edit one or a combination of: `hidden`, `sms priority`, `email priority` - Save the category
5. Note that the new values are properly reflected in the category list table.

### CODE REVIEW

File: app/notify_client/template_category_api_client.py [hunk 1]
```diff
@@ -1,110 +1,54 @@
...
+
+        if len(categories) > 0:
+            if sort_key and sort_key in categories[0]:
+                categories.sort(key=lambda category: category[sort_key].lower())
+            return categories
+        else:
+            return []
+
+    @cache.delete("template_category-{template_category_id}")
+    def update_template_category(
```
Comment (andrewleith) [inline app/notify_client/template_category_api_client.py:35]: When I update a TC, i see the changes reflected in the form.  However, when I go back to the list the changes are not there.  I think we need to also delete the `template_categories` for this to work as expected.

File: tests/app/notify_client/test_template_category_api_client.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    mock_get.assert_called_once_with(url="/template-category")
+    mock_redis_get.assert_called_once_with("template_categories")
+    mock_redis_set.assert_called_once_with(
+        "template_categories",
+        "[1, 2, 3]",
+        ex=604800,
+    )
+
+
+def test_update_template_category(template_category_client, mocker):
```
Comment (andrewleith) [inline tests/app/notify_client/test_template_category_api_client.py:78]: Are we able to test that the cache is removed here?

**Review by andrewleith — CHANGES_REQUESTED** [review]
Looks good overall.  One issue when trying to add a new category, and one cache issue when trying to update a category - both detailed below.

File: app/notify_client/template_category_api_client.py (hunk 1 above)
Comment (whabanks) [reply app/notify_client/template_category_api_client.py:37]: Good catch, I added a delete on `template_categories` when we update. I also added it when creating a category - the list page was not updating correctly when adding a new category.

Comment (andrewleith) [reply app/notify_client/template_category_api_client.py:37]: Looks good - working now!

**Review by andrewleith — APPROVED** [review]

**Pull request merged**